udata/migrations/2024-11-19-keep-only-local_authority-if-also-public_service-organization-badges.py

```diff
@@ -0,0 +1,22 @@
+"""
+This migration keeps only the "Local authority" badge if the organization also has the "Public service" badge.
+"""
+
+import logging
+
+from mongoengine.queryset.visitor import Q
+
+from udata.core.organization.constants import LOCAL_AUTHORITY, PUBLIC_SERVICE
+from udata.models import Organization
+
+log = logging.getLogger(__name__)
+
+
+def migrate(db):
+    log.info("Processing Organizations...")
+    query = Q(badges__kind=LOCAL_AUTHORITY) & Q(badges__kind=PUBLIC_SERVICE)
+
+    count = Organization.objects(query).update(set__badges__S__kind=LOCAL_AUTHORITY)
```

Comment:
I'm not sure the `__S__` will work as expected since we have two queries badges__kind?
In any case, it will replace the `PUBLIc_SERVICE` badge by another `LOCAL_AUTHORITY` leaving the orga with two `LOCAL_AUTHORITY` badges right? They don't seem to get de-duplicated

Reply:
Yes indeed, I don't know what i've done to test this out and found it working, ... I guess my query was wrong, I indeed wasn't having any org with both the badges, but they did have their badges overwritten with `local-authority`, which made them have it twice... dumb :/